allennlp/nlp_api.py

```diff
@@ -0,0 +1,289 @@
+        that knows how to process the whole list of inputs, returning a single ``Tensor``.
+
+        This abstraction is key to allowing you to easily switch between word embeddings and
+        character-level encoders, or token-in-context embeddings produced by some pre-trained
+        model, or any combination of these.  Instead of instantiating your own embedding layers, or
+        character-level CNNs, you should call :func:`embed_text_field` to do that for you, so it is
+        easy to try new representations later.  See the documentation of that function for a little
+        more detail.
+    recurrent_layers : ``Dict[str, torch.nn.Module]``, optional (default=``{}``)
+        A mapping from names to recurrent layers, like LSTMs and GRUs.  The name here corresponds
+        to some part of your `model`, like "passage_encoder", not the `type` of the recurrent
+        layer.
+    default_recurrent_layer : ``Callable[[], torch.nn.Module]``, optional (default=``None``)
+        A function that returns a new recurrent layer.  Giving this makes it so you can use unique
+        names for all recurrent layers in your model, but still only have to specify one layer
+        type.  See :func:`get_recurrent_layer` for more detail.
+    recurrent_cells : Dict[str, torch.nn.Module], optional (default=``{}``)
+        A mapping from names to recurrent cells, for building, e.g., a TreeLSTM.  Similarly to
+        ``recurrent_layers``, the names correspond to places in your model where these cells get
+        used, not to anything about what kind of cell you're using.
+    default_recurrent_cell : ``Callable[[], torch.nn.Module]``, optional (default=``None``)
+        A function that returns a new recurrent cell.  Giving this makes it so you can use unique
+        names for all recurrent cells in your model, but still only have to specify one cell type.
+        See :func:`get_recurrent_cell` for more detail.
+    """
+    def __init__(self,
+                 embedders: Dict[str, Union[torch.nn.Module, List[torch.nn.Module]]] = None,
+                 recurrent_layers: Dict[str, torch.nn.Module] = None,
+                 default_recurrent_layer: Callable[[], torch.nn.Module] = None,
+                 recurrent_cells: Dict[str, torch.nn.Module] = None,
+                 default_recurrent_cell: Callable[[], torch.nn.Module] = None):
+        if embedders is None:
+            embedders = {}
+        self._embedders = embedders
+        if recurrent_layers is None:
+            recurrent_layers = {}
+        self._recurrent_layers = recurrent_layers
+        self._default_recurrent_layer = default_recurrent_layer
+        if recurrent_cells is None:
+            recurrent_cells = {}
+        self._recurrent_cells = recurrent_cells
+        self._default_recurrent_cell = default_recurrent_cell
+
+    def embed_text_field(self,
+                         input_tensors: List[torch.Tensor],
+                         name: str = "default",
```

Comment:
Why not?  The reason this has a default and the others don't is that you pretty much always want to embed your input, and you pretty much always want to use the same embedding for every text input that you have.  There are times when you don't, which is why this is even a parameter, but most of the time, just calling `nlp_api.embed_text_field(inputs)` should be enough.